### PR TITLE
Fastify example dep update

### PR DESCRIPTION
**Pull request by crutchcorn**
This merge request updates `fastify` package from v1 to v2 and adds the `cross-env` package so `NODE_ENV=` can work on Windows as well as OSX/Linux. Also adds example build output to GITIGNORE

Worth mentioning that pressing `a` will bring up a page that says `b` and vice-versa, but this seems to be the logic from the previous example as well. Can change if wanted

### CODE REVIEW

**Review by kheruc — COMMENTED** [review]
The `"dev": "node server.js"` script should stay as it was, otherwise this Fastify server won't run when in dev mode. 

Although I can see that this whole example is not really useful, the server.js file doesn't do anything. The original author of this example implemented the server.js file to show how to use custom routing by displaying contents of `/pages/a.js` when visiting route `/b` and the opposite. Later some other contributor changed this for ssr routing in #6382, but kept client side routing as it was. Now this example shows reversed pages with client side routing, but normal pages on SSR. 

Ideally, it would be best to improve the "custom server" examples to show how to implement dynamic routing, which is the main scenario where using a custom server.js is recommended.

cc @timneutkens

**Comment by crutchcorn** [comment]
Yup, just learned about that one this morning (from Spectrum comment), apologies for not understanding sooner.

I'd be willing to work more on that if that's something that'd help this example.

I am originally not wanting to change too much in terms of logic in order to spark discussion (and admittedly to learn more regarding the intended logic as I'm unfamiliar with many of these exampes), so apologies if the `dep-updates` PRs coming tend to follow the same suit.

**Comment by timneutkens** [comment]
@crutchcorn could you send me a message at https://twitter.com/timneutkens or https://spectrum.chat/users/timneutkens 🙏

**Comment by crutchcorn** [comment]
Done via Twitter (https://twitter.com/crutchcorn)

File: examples/custom-server-fastify/package.json [hunk 1]
```diff
@@ -2,14 +2,17 @@
   "name": "custom-server-fastify",
   "version": "1.0.0",
   "scripts": {
-    "dev": "node server.js",
+    "dev": "next",
```
Comment (timneutkens) [inline examples/custom-server-fastify/package.json]: This shouldn't be changed, it won't use the custom server this way.

File: .gitignore [hunk 2]
```diff
@@ -23,3 +23,6 @@ test/**/out
 
 # Editors
 **/.idea
+
+# example output
+examples/**/out
```
Comment (Timer) [inline .gitignore]: Shouldn't this be specific to the example? 🤔

Comment (crutchcorn) [reply .gitignore]: I can make this specific to the examples in particular, but it seemed like almost all the examples I was working with had the output folder as `out`, but that's admittedly a fairly dangerous assumption. 

There doesn't seem to be any `.gitignore` files being used in any examples currently, should we start using them in order to use example specific gitignores?

File: examples/custom-server-fastify/package.json (hunk 1 above)
Comment (timneutkens) [reply examples/custom-server-fastify/package.json]: @crutchcorn could you make this change then I'll merge 🙏

Comment (timneutkens) [reply examples/custom-server-fastify/package.json]:
```suggestion
    "dev": "node server.js",
```

**Review by timneutkens — APPROVED** [review]

**Comment by timneutkens** [comment]
Thank you @crutchcorn!

**Pull request merged**